### PR TITLE
Improve AnimalOverlay (sounds and config validation)

### PR DESCRIPTION
**Pull request by sladomic**

### CODE REVIEW

File: app/src/main/java/org/literacyapp/authentication/DetectionHelper.java [hunk 1]
```diff
@@ -17,13 +17,17 @@
...
-        Point frameTopLeft = new Point(animalOverlay.getFrameStartX(), animalOverlay.getFrameStartY());
-        Point frameBottomRight = new Point(animalOverlay.getFrameEndX(), animalOverlay.getFrameEndY());
-        Rect frame = new Rect(frameTopLeft, frameBottomRight);
-        if (face.tl().x >= frame.tl().x && face.tl().y >= frame.tl().y && face.br().x <= frame.br().x && face.br().y <= frame.br().y){
-            return true;
+        if (animalOverlay != null){
+            Point frameTopLeft = new Point(animalOverlay.getFrameStartX(), animalOverlay.getFrameStartY());
+            Point frameBottomRight = new Point(animalOverlay.getFrameEndX(), animalOverlay.getFrameEndY());
+            Rect frame = new Rect(frameTopLeft, frameBottomRight);
+            if (face.tl().x >= frame.tl().x && face.tl().y >= frame.tl().y && face.br().x <= frame.br().x && face.br().y <= frame.br().y){
```
Comment (jo-elimu) [inline app/src/main/java/org/literacyapp/authentication/DetectionHelper.java]: Add parenthesises to improve readability: `((face.tl().x >= frame.tl().x) && ...`

File: app/src/main/java/org/literacyapp/authentication/animaloverlay/AnimalOverlayHelper.java [hunk 2]
```diff
@@ -82,12 +89,45 @@ public void addOverlay(Mat imgRgba){
             String assetName = animalTemplateFile.getName();
             if (animalOverlaysMap.getAnimalOverlays().get(assetName) != null){
                 int imageWidth = animalOverlaysMap.getAnimalOverlays().get(assetName).getImageWidth();
+                if (imageWidth <= 0){
```
Comment (jo-elimu) [inline app/src/main/java/org/literacyapp/authentication/animaloverlay/AnimalOverlayHelper.java:92]: Will the image width ever be <= 0?

**Review by jo-elimu — APPROVED** [review]

File: app/src/main/java/org/literacyapp/authentication/animaloverlay/AnimalOverlayHelper.java (hunk 2 above)
Comment (sladomic) [reply app/src/main/java/org/literacyapp/authentication/animaloverlay/AnimalOverlayHelper.java:92]: should not..but it's a validation for the json file :)

File: app/src/main/java/org/literacyapp/authentication/DetectionHelper.java (hunk 1 above)
Comment (sladomic) [reply app/src/main/java/org/literacyapp/authentication/DetectionHelper.java]: done

**Pull request merged**